source/dotnet/Library/AdaptiveCards.Rendering.Wpf/AdaptiveRenderContext.cs

```diff
@@ -248,5 +254,21 @@ public void ToggleVisibility(IEnumerable<AdaptiveTargetElement> targetElements)
                 }
             }
         }
+
+        public void ToggleShowCardVisibility(Button uiAction)
+        {
+            FrameworkElement card = ActionShowCards[uiAction];
+            if (card != null)
```

Comment:
>      if (card != null) [](start = 6, length = 23)

When can this be null? If we are handling it do we actually need to log/trace out (for debugging in particular) that the input was null? On the other hand, if it's an assertion like condition then just remove the null check? Though i am observing the former pattern where we check for non-null which is great but then i am curious why it would ever be null and what do we do about it in teh code?

Reply:
I think it should never be null, I'm just placing this in case it ever is null (but it should't happen)

---
In reply to: [272767915](https://github.com/Microsoft/AdaptiveCards/pull/2625#discussion_r272767915) [](ancestors = 272767915)